test/service/test_response.py

```diff
@@ -3615,5 +3631,47 @@ def test(self):
         }, response.json())
 
 
+@mock.patch.object(SourceService,
```

Comment:
Should make use of `patch_source_cache`, potentially parameterizing it. There are ways to avoid having to add `()` in every call site. I can explain in PL.